src/dosext/mfs/lfn.c

```diff
@@ -1342,3 +1342,75 @@ int mfs_lfn(void)
 		CARRY;
 	return ret;
 }
+
+static int mfs_remap_extended_open_(void)
+{
+  char *src = MK_FP32(_DS, _SI);
+  uint8_t dosint;
+  int drive, found;
+  char fpath[PATH_MAX];
+  struct stat st;
+
+  d_printf("SFN: extended open '%s'\n", src);
+
+  drive = build_posix_path(fpath, src, 0);
+  if (drive < 0)
+    return 0;
+
+  found = find_file(fpath, &st, drive, NULL);
+  d_printf("SFN: extended open '%s' does %sexist\n", fpath, found ? "" : "not ");
+
+  if ((_DX & 0b00000011) && !found)   // fail if doesn't exist
+    return lfn_error(FILE_NOT_FOUND); // 0b01 open // 0b10 truncate
+
+  if ((_DX & 0b00010000) && found)    // fail create if exists
+    return lfn_error(FILE_ALREADY_EXISTS);
+
+  dosint = (_AL == 1) ? 0x3c : 0x3d;  // select create / open function
```

Comment:
Maybe `dosfunc`?

Reply:
Do you mean you want to rename the `dosint` variable to `dosfunc`?

Reply:
Yes, dosint is 0x21.